### ISSUE TITLE
Artifacts on overviews made with cubic resampling algorithm

### ISSUE THREAD

**Issue by drnextgis** [OSGeo/gdal#3871]
In the mailing list I found a [report](https://lists.osgeo.org/pipermail/gdal-dev/2018-May/048596.html) describing the same behavior I'm experiencing. I have a raster with a mask:
```
$ gdalinfo input.tiff 
Driver: GTiff/GeoTIFF
Files: input.tiff
Size is 6845, 10302
...
Band 1 Block=6845x1 Type=Float64, ColorInterp=Gray
  NoData Value=0
  Mask Flags: PER_DATASET

$ gdalinfo -stats -json input.tif | jq '.bands[0].metadata[""].STATISTICS_MAXIMUM'
"92"
```

But when I build overviews with `cubic` resampling method I see some artifacts on the border of the overviews:

```
$ gdaladdo -r cubic input.tif 2 4 8 16 32 64
$ gdal_translate -oo OVERVIEW_LEVEL=5 input.tif gdal-cubic-ovr-5.tif
$ gdalinfo -stats -json gdal-cubic-ovr-5.tif | jq '.bands[0].metadata[""].STATISTICS_MAXIMUM'
"9329.2841796875"
```

Original raster and resampled one:

![Untitled Project](https://user-images.githubusercontent.com/866124/119118177-40979200-ba2a-11eb-9599-0b08fef837ba.png)

I'm not sure if it is an expected thing for `cubic` resampling or a bug. For `nearest` method everything works fine.

**Comment by rouault** [comment]
Please attach a minimum reproducing dataset

**Comment by drnextgis** [comment]
@rouault thank you for your prompt reply. I've sent you the dataset via email.

**Comment by drnextgis** [comment]
[input.tif.zip](https://github.com/OSGeo/gdal/files/6541296/input.tif.zip)

**Comment by jratike80** [comment]
What is your GDAL version?

**Comment by jratike80** [comment]
With GDAL 3.3.0 on Windows I did:
```
gdaladdo -r cubic input.tif 2 4 8 16 32 64
gdalinfo input.tif -stats
Max: 255
gdal_translate -oo OVERVIEW_LEVEL=5 input.tif gdal-cubic-ovr-5.tif
gdalinfo -stats gdal-cubic-ovr-5.tif
Max: 53789.26953125
```

**Comment by drnextgis** [comment]
> What is your GDAL version?

```
$ gdalinfo --version
GDAL 3.3.0, released 2021/04/26
```

**Comment by drnextgis** [comment]
I believe we can remove "awaiting_feedback" since @jratike80 was able to reproduce the issue.

**Comment by drnextgis** [comment]
I thought that maybe this issue and https://github.com/OSGeo/gdal/issues/4000 have the same root cause. But apparently they don't. Still can reproduce the issue with `osgeo/gdal:alpine-small-latest`.

**Comment by rouault** [comment]
Fixed per #5313 / duplicate of #5016

$ gdalinfo -stats -json gdal-cubic-ovr-5.tif | jq '.bands[0].metadata[""].STATISTICS_MAXIMUM'
"180.42585754395"
